service/paxos/paxos_state.cc

```diff
@@ -309,7 +404,35 @@ future<column_mapping> paxos_store::get_column_mapping(table_id table_id, table_
 }
 
 future<schema_ptr> paxos_store::get_paxos_state_schema(const schema& s, db::timeout_clock::time_point timeout) const {
-    co_return db::system_keyspace::paxos();
+    if (const auto& table = s.table(); !table.uses_tablets()) {
+        co_return db::system_keyspace::paxos();
+    }
+
+    // The paxos state schema only includes the partition key from the base table.
+    // Since partition keys can't be altered via CQL, the paxos schema doesn't need to change
+    // when the base schema changes.
+    // When the LWT coordinator sends RPCs to replicas, some may not yet have the paxos schema.
+    // Wait for the schema to appear here, or throw 'no_such_column_family' if the base table was dropped.
+    // (see try_get_paxos_state_schema)
+
+    SCYLLA_ASSERT(_features.lwt_with_tablets);
```

Comment:
You can use assert(). Unless you really want us to crash here in a release build as well.